integration/service/test/test_cron_use_case_plot.py

```diff
@@ -0,0 +1,182 @@
+            for metric_name, metrics in report['metrics'].items():
+                df_row['metric-name'] = metric_name
+                for m_key, m_value in metrics.items():
+                    df_row[m_key] = m_value
+                report_list.append(df_row.copy())
+        self._df = DataFrame(report_list)
+        self._df.sort_values('sample-time-first', inplace=True)
+        if len(self._df) == 0:
+            raise RuntimeError('Could not load reports')
+        self._colors = dict(
+            mean_marker='rgb(247, 240, 25)',
+            min_marker='rgb(82, 115, 222)',
+            max_marker='rgb(217, 85, 85)',
+            std_marker='rgb(68, 68, 68)',
+            std_fill='rgba(68, 68, 68, 0.5)',
+            range_fill='rgba(68, 68, 68, 0.3)',
+            pie_line='#000000',
+            pie_gpu='red',
+            pie_cpu='orange',
+            pie_dram='yellow',
+            background='#2980b9',
+            page_text='#fcfcfc',
+        )
+
+
+    def calculate_energy(self, domain, begin_date, end_date):
+        domain = domain.upper()
+        metric_name = f'{domain}_POWER'
+        time = self.get_stat(begin_date, end_date, metric_name, 'sample-time-first')
+        time_total = self.get_stat(begin_date, end_date, metric_name, 'sample-time-total')
+        power = self.get_stat(begin_date, end_date, metric_name, 'mean-arithmetic')
+        kwh_factor = 1.0 / 3.6e6
+        # TODO include energy from last report as well
+        return kwh_factor * sum((power.shift(1) * (time - time.shift(1)).dt.total_seconds()).dropna())
+
+    def dataframe(self):
+        return self._df
+
+    def get_stat(self, begin_date, end_date, metric_name, stat_name):
+        return self._df.loc[self._df['sample-time-first'].between(begin_date, end_date) & \
+                            (self._df['metric-name'] == metric_name), stat_name]
+
+    def plot_power(self, domain, begin_date, end_date):
+        domain = domain.upper()
+        metric_name = f'{domain}_POWER'
+        sample_time_first = self.get_stat(begin_date, end_date, metric_name, 'sample-time-first')
+        sample_time_total = self.get_stat(begin_date, end_date, metric_name, 'sample-time-total')
+        record_mean_time = sample_time_first + to_timedelta(sample_time_total / 2.0, unit='s')
+        record_min_power = self.get_stat(begin_date, end_date, metric_name, 'min')
+        record_max_power = self.get_stat(begin_date, end_date, metric_name, 'max')
+        record_mean_power = self.get_stat(begin_date, end_date, metric_name, 'mean-arithmetic')
+        record_std_power = self.get_stat(begin_date, end_date, metric_name, 'std')
+        record_plus_std = [min((a, b)) for (a, b) in zip(record_mean_power + record_std_power, record_max_power)]
+        record_minus_std = [max((a, b)) for (a, b) in zip(record_mean_power - record_std_power, record_min_power)]
+        fig = go.Figure([
+            go.Scatter(
+                name=f'{domain} Power Std',
```

Comment:
I think we want to swap the order of Std and Min/Max so the min/max is shaded first